src/react-components/input/Slider.js

```diff
@@ -0,0 +1,143 @@
+import React from "react";
+import { useState, useCallback, useEffect, useRef } from "react";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+import styles from "./Slider.scss";
+import { trackColor } from "./Slider.scss";
+
+function round(step, n) {
+  return Math.round(n / step) * step;
+}
+
+function getLinearGradientCSS(ratio, leftColor, rightColor) {
+  return [
+    "-webkit-gradient(",
+    "linear, ",
+    "left top, ",
+    "right top, ",
+    "color-stop(" + ratio + ", " + leftColor + "), ",
+    "color-stop(" + ratio + ", " + rightColor + ")",
+    ")"
+  ].join("");
+}
+
+function updateGradient(el, min, max, value) {
+  if (el) {
+    const ratio = ((value - min) / (max - min)).toFixed(2);
+    el.style.backgroundImage = getLinearGradientCSS(ratio, trackColor, "transparent");
+  }
+}
+
+export function Slider({ min, max, step, digits, value, defaultValue, onChange, disabled, className }) {
+  const myRoot = useRef();
+  const inputRef = useRef();
+  const [displayValue, setDisplayValue] = useState(-1);
```

Comment:
This extra state existing is surprising. I expect we just use the value passed in directly, which will get updated because we call `onChange` every time the `range` input changes. The usecase I could see for it existing would be if we only wanted to call `onChange` once the user has stopped dragging, but we aren't currently doing that.

Reply:
All this state handling was original to deal with the double updating case where the slider itself could update it's value and also be updated by an external component from outside. As part of the refactor I've got rid of this an a lot of related code. Now the component should be much more lean and refresh only when strictly necessary.